src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
For example, one can have such git repo: `repo`. It is a perfect git+ssh repo, given it is configured in `.ssh/config`. In this case:

- scheme is not required
- user is not required
- repo name is not required (defaults to just `/`)
- `.git` is not required